addons/stock/migrations/11.0.1.1/pre-migration.py

```diff
@@ -2,21 +2,104 @@
 # Copyright 2017-2018 Tecnativa - Pedro M. Baeza
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
+from psycopg2.extensions import AsIs
 from openupgradelib import openupgrade
 
 
-def copy_global_rules(env):
-    """Copy global rules to another table and remove them from main one."""
+def _get_dummy_route_id(env, default):
+    """ Return an inactive dummy route. Create on demand """
+    if default:
+        return default
+    env.cr.execute(
+        """ INSERT INTO stock_location_route
+        (create_date, create_uid, name, active)
+        VALUES(now() at time zone 'UTC', 1,
+        'OpenUpgrade 11.0 Dummy Route for inactive rules & paths',
+        False) RETURNING id """)
+    return env.cr.fetchone()[0]
+
+
+@openupgrade.logging()
+def create_specific_procurement_rules_from_globals(env):
+    """Procurement rules and location paths without a route were considered
+    global in Odoo 10. Now that the route_id is a required field in both
+    tables, expand active rules and paths without a route so that we will end
+    up with one record for each route. Don't expand inactive records or expand
+    records for inactive routes. If necessary, create a dummy inactive route
+    to assign to inactive records without a route.
+
+    Note that rules and paths are left dangling if you unlink their route.
+    Before starting the migration, check to see which of your 'global'
+    rules need to be set to inactive to prevent the unexpected expansion
+    of such rules and paths.
+    """
+    dummy_route_id = False
+    env.cr.execute(
+        "SELECT id FROM stock_location_route WHERE active ORDER BY id")
```

Comment:
shouldn't you first order by [`sequence`](https://github.com/OCA/OpenUpgrade/blob/10.0/addons/stock/models/stock_location.py#L118)?

Reply:
Ah, the ordering is a remainder from an earlier attempt in which I needed to know in the post script which route I assigned in the pre script. It is arbitrary but in the current version I might just as well leave it out (and I might just as well leave it in, which I think I will).

Reply:
But I will add a limit

Reply:
Added a limit, removed the ordering after all and added a clarifying comment.